app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,90 @@
+class OrdersController < ApplicationController
+  include CurrentCart
+  before_action :set_cart, only: [:new, :create]
+  before_action :ensure_cart_has_items, only: :new
+  before_action :set_order, only: [:show, :edit, :update, :destroy]
+
+  # GET /orders
+  # GET /orders.json
+  def index
+    @orders = Order.all.order(id: :desc)
+    render "orders/index.json"
+  end
+
+  # GET /orders/1
+  # GET /orders/1.json
+  def show
+  end
+
+  # GET /orders/new
+  def new
+    @order = Order.new
+  end
+
+  # GET /orders/1/edit
+  def edit
+  end
+
+  # POST /orders
+  # POST /orders.json
+  def create
+    @order = Order.new(order_params)
+    @order.add_line_items_from_cart(@cart)
+
+    respond_to do |format|
+      if @order.save
+        Cart.destroy(session[:cart_id])
+        session[:cart_id] = nil
+        flash[:notice] = 'Thank you for your order!'
```

Comment:
You could probably extract this into a private method on the controller to make this a bit more of a succinct process. Something like `reset_cart_after_order` that just contains these three lines. That's strictly an open statement though.